src/lib/optimizer/chunk_statistics/range_filter.hpp

```diff
@@ -49,7 +70,17 @@ class RangeFilter : public AbstractFilter {
 };
 
 template <typename T>
-std::unique_ptr<RangeFilter<T>> RangeFilter<T>::build_filter(const pmr_vector<T>& dictionary) {
+std::unique_ptr<RangeFilter<T>> RangeFilter<T>::build_filter(const pmr_vector<T>& dictionary,
+                                                             uint32_t max_ranges_count) {
+  static_assert(std::is_arithmetic_v<T>, "Range filters are only allowed on arithmetic types.");
+  DebugAssert(!dictionary.empty(), "The dictionary should not be empty.");
+
+  if (dictionary.size() == 1) {
+    std::vector<std::pair<T, T>> ranges;
+    ranges.emplace_back(dictionary.front(), dictionary.front());
+    return std::make_unique<RangeFilter<T>>(std::move(ranges));
+  }
+
   // calculate distances by taking the difference between two neighbouring elements
   std::vector<std::pair<T, size_t>> distances;
```

Comment:
we should add a comment explaining the data format of this vector. it was quite difficult to re-understand this when I looked over the code.

Reply:
something like `stores <distance to next element, dictionary index>`